src/monitoring/index.ts

```diff
@@ -65,9 +67,74 @@ export default class MonitoringService {
     module: "monitoringService",
   })
 
+  private readonly REDIS_CONNECTION = {
+    host: config.get("bullmq.redisHostname"),
+    port: 6379,
+  }
+
+  private readonly queue = new Queue("MonitoringQueue", {
+    connection: {
+      ...this.REDIS_CONNECTION,
+    },
+    defaultJobOptions: {
+      removeOnComplete: true,
+      removeOnFail: true,
+      attempts: 3,
+      backoff: {
+        type: "exponential",
+        delay: 60000, // this operation is not critical, so we can wait a minute
+      },
+    },
+  })
+
+  private readonly worker: Worker<unknown, string, string> | undefined
+
   constructor({ launchesService }: MonitoringServiceInterface) {
     autoBind(this)
+    const jobName = "dnsMonitoring"
     this.launchesService = launchesService
+    this.worker = new Worker(
+      this.queue.name,
+      async (job: Job) => {
+        this.monitoringServiceLogger.info(`Monitoring Worker ${job.id}`)
+        if (job.name === jobName) {
+          // The retry's work on a thrown error, so we need to convert the neverthrow to a promise
+          const res = await convertNeverThrowToPromise(this.driver())
```

Comment:
just off the top of my head, this seems odd. we're defining driver in `MonitoringService` but actually `Worker` uses it. This seems to blur the lines between the service and the worker. 

i think this PR might be better if we split the concerns here! logically speaking, i'd expect the worker to be the one to actually _do_ the work, and for the monitoring service to just handle the worker scheduling/linking up the worker to the queue. 

i'm not sure if this delineation makes sense (it might not) but i think having a method defined on the service that's **only called by the worker** is definitely a code smell

Reply:
hey, this has been refactored